src/ray/gcs/gcs_server/test/redis_gcs_table_storage_test.cc

```diff
@@ -0,0 +1,506 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+#include "ray/common/test_util.h"
+#include "ray/gcs/gcs_server/gcs_table_storage.h"
+#include "ray/gcs/store_client/redis_store_client.h"
+#include "ray/gcs/store_client/test/store_client_test_base.h"
+
+namespace ray {
+
+class GcsTableStorageTest : public gcs::StoreClientTestBase {
+ public:
+  GcsTableStorageTest() {}
+
+  virtual ~GcsTableStorageTest() {}
+
+  void InitStoreClient() override {
+    gcs::RedisClientOptions options("127.0.0.1", REDIS_SERVER_PORT, "", true);
+    redis_client_ = std::make_shared<gcs::RedisClient>(options);
+    RAY_CHECK_OK(redis_client_->Connect(io_service_pool_->GetAll()));
+
+    gcs_table_storage_ = std::make_shared<gcs::RedisGcsTableStorage>(redis_client_);
+  }
+
+  void DisconnectStoreClient() override { redis_client_->Disconnect(); }
+
+ protected:
+  rpc::JobTableData GenJobTableData(JobID job_id) {
```

Comment:
These `GenXTableData` methods are duplicate with those in `src/ray/gcs/gcs_server/test/gcs_server_rpc_test.cc`. Can you move them to a common util file?

Reply:
fixed